core/chain-configs/src/genesis_config.rs

```diff
@@ -134,6 +138,28 @@ pub struct GenesisConfig {
     pub minimum_stake_divisor: u64,
 }
 
+impl From<&GenesisConfig> for EpochConfig {
+    fn from(config: &GenesisConfig) -> Self {
```

Comment:
@matklad here I am not sure how to avoid abusing `From`. `EpochConfig` is defined in a dependency of this crate (where `GenesisConfig`) is defined and unless we do some larger refactoring, it seems that there is no easy way to get around this.

Reply:
Yeah, it is a slight abuse, but I think it is fine (I wonder if this should become idiomatic at some point). Alternatively, you can do:

1. implement `to_epoch_config(&self)` on the `GenesisConfig`
2. implement a new trait like:

```rust
trait NewEpochConfig {
    fn from_genesis_config(&GenesisConfig) -> EpochConfig
}

impl NewEpochConfig for EpochConfig {
    fn from_genesis_config(&GenesisConfig) -> EpochConfig {
       ...
    }
}
```

note: you will need to import this trait where you want to call `from_genesis_config` on `EpochConfig` value.

P.S. we use [easy-ext](https://docs.rs/easy-ext/) to shink the duplication

Reply:
```rust
impl GenesisConfig {
    fn [to_]epoch_config(&self) -> EpochConfig
}
```

is how I'd personally write this. I would definitely stay away from using extension traits, they seem to be pure accidental complexity for this use-case.

But, again, it's not that From impl is wrong :)